pkg/arvo/app/btc-node-hook.hoon

```diff
@@ -0,0 +1,373 @@
+        btc-core  +>
+        bc        ~(. btc-core bowl)
+        def       ~(. (default-agent this %|) bowl)
+    ::
+    ++  on-init
+      ^-  (quip card _this)
+      :-  ~
+      %_  this
+        endpoint  ''
+        headers   ~
+      ==
+    ::
+    ++  on-save   !>(state)
+    ++  on-load
+      |=  old=vase
+      `this(state !<(state-zero old))
+    ::
+    ++  on-poke
+      |=  [=mark =vase]
+      ^-  (quip card _this)
+      |^  ?+    mark    (on-poke:def mark vase)
+              %btc-node-hook-action
+            (hook-action !<(btc-node-hook-action vase))
+          ::
+              %btc-node-hook-command
+            (hook-command !<(btc-node-hook-command vase))
+          ==
+      ::
+      ++  hook-action
+        |=  act=btc-node-hook-action
+        ^-  (quip card _this)
+        =/  body=request:rpc:jstd
+          (request-to-rpc:btc-rpc:lib act)
+        =/  req=request:http
+          :*  %'POST'
+              (endpoint-url:bc act)
+              headers
+              =,  html
+              %-  some
+                %-  as-octt:mimes
+                  (en-json (request-to-json:rpc:jstd body))
+          ==
+        =/  out  *outbound-config:iris
+        :_  this
+        [%pass /[(scot %da now.bowl)] %arvo %i %request req out]~
+      ::
+      ++  hook-command
+        |=  comm=btc-node-hook-command
+        ^-  (quip card _this)
+        ?+    -.comm  ~|  [%unsupported-hook-command -.comm]  !!
+            %credentials
+          :_  this(endpoint url.comm, headers heads.comm)
+          [%pass / %arvo %d %flog [%text "credentials updated..."]]~
+        ==
+      --
+    ::
+    ++  on-watch  on-watch:def
+    ++  on-leave  on-leave:def
+    ++  on-peek   on-peek:def
+    ++  on-agent  on-agent:def
+    ++  on-arvo
+      |=  [=wire =sign-arvo]
+      ^-  (quip card _this)
+      =*  response  client-response.sign-arvo
+      =^  cards  state
+        ?+    +<.sign-arvo    (on-arvo:def wire sign-arvo)
+            %http-response    (http-response:bc wire response)
+        ==
+      [cards this]
+    ::
+    ++  on-fail   on-fail:def
+    --
+::
+|_  =bowl:gall
+::
+++  httr-to-rpc-response
+  |=  hit=httr:eyre
+  ^-  response:rpc:jstd
+  ~|  hit
+  :: ?.  ?=(%2 (div p.hit 100))
+  ::   fail+hit
+  =/  jon=json  (need (de-json:html q:(need r.hit)))
+  =,  dejs-soft:format
+  ^-  response:rpc:jstd
+  =;  dere
+    =+  res=((ar dere) jon)
+    ?~  res  (need (dere jon))
+    [%batch u.res]
+  |=  jon=json
+  ^-  (unit response:rpc:jstd)
+  =/  res=[id=(unit @t) res=(unit json) err=(unit json)]
+    %.  jon
+    =,  dejs:format
+    =-  (ou -)
+    :~  ['id' (uf ~ (mu so))]
+        ['result' (uf ~ (mu same))]
+        ['error' (uf ~ (mu same))]
+    ==
+  ?:  ?=([^ * ~] res)
+    `[%result [u.id.res ?~(res.res ~ u.res.res)]]
+  ~|  jon
+  ::%-  need
+  :: ((ot id+so error+(ot code+no message+so ~) ~) jon)
+  `[%error (handle-error jon)]
+::
+++  handle-error
+  |=  =json
+  ?~  json  ['' '' '']
```

Comment:
Would love a typecast here. Maybe just include the `%error` tag, and cast to `$>(%error response:rpc:jstd)`.

(It's really more like `+parse-error`, isn't it? Kinda surprised we don't have a generic parser for rpc responses in the stdlib. Clostest thing is [this in /lib/ethio](https://github.com/urbit/urbit/blob/6ef08962efa5aef197fdf961261e918075491d13/pkg/arvo/lib/ethio.hoon#L107-L119).)